tools/xtce_generation/telemetry_generator.py

```diff
@@ -250,9 +250,19 @@ def create_remaining_parameters(self, parameter_set):
 
             description = Et.SubElement(parameter, "xtce:LongDescription")
 
-            description.text = (
-                row.get("shortDescription") or row.get("longDescription") or ""
+            # handle nan floating point values and replace with empty strings
```

Comment:
I know `short_description` and `long_description` has been causing issues.

Reply:
You mean using those variable names were causing issues before?

Reply:
What you have is good. I mean to say thank you for adding these checks. I thought mine will take care this issue but looks like it didn't. I should have tested better.

Reply:
Ah, gotcha. No problem! Hopefully these changes don't cause issues for anyone.

Reply:
Is this actually an issue upstream of here, so it actually needs the fix to be in the parser somewhere? I would have expected `row.get("shortDescription")` to always return a string, and never a floating point NaN, so I'm kind of curious if there is something more fundamental about this happening that might cause issues in other places too.

Reply:
I think the reason I'm getting the NaNs is when pandas parses the empty cells, it makes it a NaN. I just assumed other excel sheets had different formatting that was causing the empty cells to be parsed as None since no one else was having this issue. We could do a `pkt.fillna("None")` to get rid of all NaNs and then just check for `"None"` string, but if it's parsing empty cells from other sheets as `None`, then we'd still need to handle that string (I don't think you can fillna with None type)

Reply:
Ahh, that makes sense, and it is an entire column of empty items is my guess and pandas infers we are dealing with numerical types of possible. Perhaps we could do .astype(str) or some other way of setting the dtype on those specific columns to make sure they are parsed as strings and not floats?

Reply:
Looks like there is a dtype option to read_csv for specific columns we could specify when initially reading the file in...

https://pandas.pydata.org/docs/reference/api/pandas.read_csv.html

Reply:
I think we need to use `pandas.ExcelFile.parse` since we're reading sheets in an excel file which doesn't have `dtype` option. There's a `converters` option to convert a column to a certain type, but that's not changing the type for NaNs when I try `converters={"longDescription":str}`. There's a `dtype_backend=_NoDefault.no_default`, but I'm not finding much info on what that does. I might be missing something though. It's been awhile since I've used pandas.

Reply:
And yeah, I can do the `.astype(str)` and check for a `"nan"` string (I didn't see that comment when I responded previously). I can try just handling NaN and see if I get any issues generating xtce for SWE or CODICE, but we'll still need to check for both if others are parsing as `None` for whatever reason.

Reply:
OK, interesting that HIT is the only instrument with blank cells to cause this! This definitely seems like something others may produce in the future as well...

What about adding a converter like this? https://stackoverflow.com/questions/32591466/python-pandas-how-to-specify-data-types-when-reading-an-excel-file

May be splitting hairs here though, so your current solution is good, it just feels like there should be a way to push this upstream a bit further as well.

Reply:
I figured out what's going on. An empty cell in a column is parsed as NaN, but if row.get() tries to access data from a column that doesn't exist it returns `None`. SWE doesn't have a `shortDescription` column which is where the `None` was coming from before. Would it be better to specify column names needed by `pd.ExcelFile.parse("packet name", usecols=["mnemonic", "startByte", ...] )` and return an error saying the spreadsheet should be updated with the missing column?

Reply:
That makes sense now, nice investigation!

> Would it be better to specify column names needed by pd.ExcelFile.parse("packet name", usecols=["mnemonic", "startByte", ...] ) and return an error saying the spreadsheet should be updated with the missing column? That could also be helpful in catching future spreadsheet issues, but it might be more annoying than it's worth.

I don't think we want to force people to add short and long descriptions if they aren't in the spreadsheets, that sounds like a lot of headache like you say. Yet another option to consider would be to return `np.nan` from the get lookup fallback:

```python
if not np.isnan(row.get("longDescription", np.nan)):
    ...
```
So then we don't have to check for both nan and None. Also see my comment below about adding both short and long descriptions as two separate if clauses, which might simplify some of the logic here as well if you want to implement that.

Reply:
Good idea! Thanks! I also just found out that `pd.isna()` / `pd.notna()` handles NaN and None the same.